python/pyspark/sql/tests/test_dataframe.py

```diff
@@ -1138,6 +1138,39 @@ def test_create_nan_decimal_dataframe(self):
             [Row(value=None)],
         )
 
+    def test_transform_param(self):
+        def transform_fun(df, m, *, param):
+            return df.filter(col("c1") <= param).withColumn("c1", col("c1") * m)
+
+        def assert_equals(observed):
+            actual = observed.collect()
+            self.assertEqual(
+                [
+                    {
+                        "c1": 3,
+                    },
+                    {
+                        "c1": 6,
+                    },
+                ],
+                [row.asDict() for row in actual],
+            )
+
+        sdf = self.spark.createDataFrame([(1,), (2,), (3,)], ["c1"])
+        assert_equals(sdf.transform(transform_fun, 3, param=2))
+        assert_equals(sdf.transform(lambda df: transform_fun(df, 3, param=2)))
+        from functools import partial
```

Comment:
nit: I'm not strong feeling about this, but can we move this import to the top of file as PEP recommends ??